artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AckManager.java

```diff
@@ -149,15 +149,32 @@ public boolean initRetry() {
 
       HashMap<SimpleString, LongObjectHashMap<JournalHashMap<AckRetry, AckRetry, Queue>>> retries = sortRetries();
 
+      scanAndFlushMirrorTargets();
+
       if (retries.isEmpty()) {
          logger.trace("Nothing to retry!, server={}", server);
          return false;
       }
 
-      progress = new MultiStepProgress(sortRetries());
+      progress = new MultiStepProgress(retries);
       return true;
    }
 
+   private void scanAndFlushMirrorTargets() {
+      logger.debug("scanning and flushing mirror targets");
+      // this will navigate on each connection, find the connection that has a mirror controller, and call flushMirrorTarget for each MirrorTargets. (it should be 1 in most cases)
+      // An alternative design instead of going through the connections, would be to register the MirrorTargets within the AckManager, however to avoid memory leaks after disconnects and reconnects it is safer to
+      // scan through the connections
+      server.getRemotingService().getConnections().stream().
+         filter(c -> c instanceof ActiveMQProtonRemotingConnection && ((ActiveMQProtonRemotingConnection) c).getAmqpConnection().getMirrorControllerTargets() != null).
+         forEach(c -> ((ActiveMQProtonRemotingConnection) c).getAmqpConnection().getMirrorControllerTargets().forEach(this::flushMirrorTarget));
```

Comment:
This might work for now, but anyone with lots of connections seems likely to find this causes a noticable amount of CPU usage. Per other comment, its also already a leak of sorts.